h/h_api/bulk_api/entry_point.py

```diff
@@ -89,3 +97,31 @@ def _commands_from_ndjson(lines):
 
             # Try catch JSON errors here
             yield CommandBuilder.from_data(data)
+
+    @staticmethod
+    def string_to_lines(string):
+        return (line for line in string.strip().split("\n"))
+
+    @staticmethod
+    def bytes_to_lines(stream, chunk_size=16384):
+        line = b""
+
+        while True:
+            block = stream.read1(chunk_size)
```

Comment:
I think `read1` will only issue a single read command to the underlying buffer object, even if your chunk size is very large. What I interpret that to mean is, if there is a natural block size in the underlying handle and we have specified a larger number than that, then we'll get the smaller amount. The hope here is to not get slightly out of step with the underlying size and end-up issuing calls where the last one barely reads anything.